WebRoot/js/zimbraAdmin/globalconfig/view/GlobalConfigXFormView.js

```diff
@@ -329,60 +329,57 @@ GlobalConfigXFormView.updateRetentionPolicy = function(form) {
     }
 }
 
-GlobalConfigXFormView.suspendSyncListener = function() {
+GlobalConfigXFormView.registerDeviceListener = function(operation) {
     var form = this.getForm()
     var obj = form.getInstanceValue(ZaGlobalConfig.A2_registeredDevice_Selection)[0];
 
-    var result = ZaRegisterDevice.quarantineDevice(obj)[0];
-    GlobalConfigXFormView.updateRegisteredDeviceList(obj, result, form);
-}
-
-GlobalConfigXFormView.removeDeviceListener = function() {
-    var selected = this.getForm().getInstanceValue(ZaGlobalConfig.A2_registeredDevice_Selection)[0];
-
-    var all = this.getForm().getInstanceValue(ZaGlobalConfig.A2_registeredDevice);
-
-    if (!selected) {
+    if (!obj) {
         return;
     }
 
-    ZaRegisterDevice.removeDevice(selected);
-    AjxUtil.arrayRemove(all, selected);
-
-    this.getForm().setInstanceValue([], ZaGlobalConfig.A2_registeredDevice);
-    this.getForm().setInstanceValue(all, ZaGlobalConfig.A2_registeredDevice);
-    this.getForm().setInstanceValue([], ZaGlobalConfig.A2_registeredDevice_Selection);
-}
-
-GlobalConfigXFormView.resumeSyncListener = function() {
-    var form = this.getForm();
-    var obj = form.getInstanceValue(ZaGlobalConfig.A2_registeredDevice_Selection)[0];
-    var result = ZaRegisterDevice.allowDeviceSync(obj)[0];
-
-    GlobalConfigXFormView.updateRegisteredDeviceList(obj,result, form);
-}
-
-GlobalConfigXFormView.updateRegisteredDeviceList = function(obj, result, form) {
-    var all = form.getInstanceValue(ZaGlobalConfig.A2_registeredDevice);
-
-    if (result && obj && obj[ZaRegisterDevice.RD_Status] !==  result[ZaRegisterDevice.RD_Status]) {
-        var index = AjxUtil.indexOf(all, obj);
-
-        AjxUtil.arrayRemove(all, obj);
-        AjxUtil.arrayAdd(all, result, index);
-
-        form.setInstanceValue([], ZaGlobalConfig.A2_registeredDevice);
-        form.setInstanceValue(all, ZaGlobalConfig.A2_registeredDevice);
-        form.setInstanceValue([], ZaGlobalConfig.A2_registeredDevice_Selection);
+    let result, statusValue ;
+
+    if(operation === 'remove') {
+        result = ZaRegisterDevice.removeDevice(obj);
+        statusValue = 0;
+    } else if (operation === 'suspend') {
+        result = ZaRegisterDevice.quarantineDevice(obj);
+    } else if (operation === 'resume') {
+        result = ZaRegisterDevice.allowDeviceSync(obj);
+    } else if (operation === 'wipe') {
+        result = ZaRegisterDevice.wipeDevice(obj);
+    } else if (operation === 'wipeCancel') {
+        result = ZaRegisterDevice.wipeCancelDevice(obj);
+    } else if(operation === 'block') {
+        result = ZaRegisterDevice.blockDevice(obj);
     }
-} 
 
-GlobalConfigXFormView.wipeDeviceListener = function() {
-    var form = this.getForm();
-    var obj = this.getForm().getInstanceValue(ZaGlobalConfig.A2_registeredDevice_Selection)[0];
-    var result = ZaRegisterDevice.wipeDevice(obj)[0];
+    if (result && result[0]) {
```

Comment:
Looks a little confusing to me, we are checking `result` for null or undefined but if it is null then we are explicitly assigning null to it. Any reason?

Reply:
the result value can be empty object in case of remove device req.,  it will have result[0] value in case of other operations OR it can be simply null if request is failed.